src/snappy_device_agents/devices/maas2/maas_storage.py

```diff
@@ -372,7 +372,10 @@ def _get_child_device(self, parent_device):
         """
         children = []
         for dev in self.device_list:
-            if dev["parent_disk"] == parent_device["id"]:
+            if (
+                "parent_disk" in dev
+                and dev["parent_disk"] == parent_device["id"]
+            ):
```

Comment:
why not just: `if dev.get("parent_disk") == parent_device["id"]`?

Reply:
Thought I had done this after the change on line 478... this is the preferred method and is more consistent (as below).